lib/git_hooks/installer.rb

```diff
@@ -1,33 +1,42 @@
 module GitHooks
   class Installer
-    attr_accessor :hook
-
     HOOK_SAMPLE_FILE = 'hook.sample'
 
-    def initialize(hook)
+    def initialize(hook, ruby_path = nil)
       @hook = hook
+      @ruby_path = ruby_path
     end
 
     def install(force = false)
-      return true if installed?
+      throw Exceptions::UnknownHookPresent.new(hook) if !force && installed?
+
+      hook_script = hook_template
```

Comment:
Got your point, and I have a few points. 

First of all, this way to validate is installed is not good enough.  Some time we could release  other hook validations, not only pre-commit. I really don't know how to validate this. 

Other point is the fact that I don't know why we need the link, do you remember some point to not move to your strategy? I don't.